src/on.js

```diff
@@ -346,56 +368,29 @@
 		return h;
 	};
 
-	INVALID_PROPS = {
-		isTrusted: 1
-	};
-	function mix (object, value) {
-		if (!value) {
-			return object;
-		}
-		if (typeof value === 'object') {
-			for(var key in value){
-				if (!INVALID_PROPS[key]) {
-					object[key] = value[key];
-				}
-			}
-		} else {
-			object.value = value;
-		}
-		return object;
-	}
-
 	on.emit = function (node, eventName, value) {
-		node = getNode(node);
-		var event = document.createEvent('HTMLEvents');
+		node = typeof node == 'string' ? getNodeById(node) : node;
+		var event = node.ownerDocument.createEvent('HTMLEvents');
 		event.initEvent(eventName, true, true); // event type, bubbling, cancelable
 		return node.dispatchEvent(mix(event, value));
 	};
 
 	on.fire = function (node, eventName, eventDetail, bubbles) {
-		var event = document.createEvent('CustomEvent');
+		node = typeof node == 'string' ? getNodeById(node) : node;
+		var event = node.ownerDocument.createEvent('CustomEvent');
 		event.initCustomEvent(eventName, !!bubbles, true, eventDetail); // event type, bubbling, cancelable, value
 		return node.dispatchEvent(event);
 	};
 
+	// TODO: undocumented and unused? why is an one-liner available as a function?
```

Comment:
This code has become less necessary since event.key became common. I still use it, but will look to factor it out.